src/run.ts

```diff
@@ -79,5 +79,33 @@ async function runSetup(): Promise<null | Error> {
   core.info(`Successfully setup buf version ${version}`);
   core.info(cp.execSync(`${binaryPath} --version`).toString());
 
+  const bufToken = core.getInput("buf_token");
+  const bufUser = core.getInput("buf_user");
+  if (bufUser !== "" && bufToken !== "") {
+    core.info(`buf_user and buf_token supplied, logging in...`);
+    core.info(
+      cp
+        .execSync(
+          `${binaryPath} registry login --username ${bufUser} --token-stdin`,
+          { input: bufToken }
+        )
+        .toString()
+    );
+  } else {
+    if (bufUser !== "") {
+      core.info(
+        `buf_user is supplied, must also supply buf_token to log into Buf registry`
+      );
+    } else if (bufToken !== "") {
+      core.info(
+        `buf_token is supplied, must also supply buf_user to log into Buf registry`
+      );
```

Comment:
Instead of a nested `if... else if` block, we could probably use early returns to handle the log statements for readability:

```
if (bufUser !== "" && bufToken !== "" {
    // the logic you already have, which is great
    return null;
}
if (bufUser !== "") {
    core.info(`buf_user is supplied, must also supply...`)
    return null;
}
if (bufToken !== "") {
    // same logic
    return null;
}
core.info(`buf_user and buf_token are not configured, not logging into Buf Schema Registry`)
return null;
```

Reply:
Thank you for bringing this up! I have updated this in a recent commit.